libbeat/common/jsontransform/jsonhelper.go

```diff
@@ -83,11 +85,15 @@ func WriteJSONKeys(event *beat.Event, keys map[string]interface{}, overwriteKeys
 				continue
 			}
 			event.Fields[k] = vstr
-
-		default:
-			event.Fields[k] = v
 		}
 	}
+
+	// We have accounted for @timestamp, @metadata, type above. So let's remove these keys and
+	// deep update the event with the rest of the keys
+	delete(keys, "@timestamp")
```

Comment:
Nit: How about using a `for` loop instead of calling `delete` three times?

```golang
for _, name := range []string{"@timestamp", "@metadata", "type"} {
    delete(keys, name)
}
```

Reply:
Addressed in 4f1ee99ec.